### PR TITLE
[quant][onnx] Support conversion of quantized sigmoid operator from pytorch to caffe2

### PR DESCRIPTION
**Pull request by supriyar**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#34629 [quant][onnx] Support conversion of quantized sigmoid operator from pytorch to caffe2**
* #33945 [quant][onnx] Add support to convert max_pool2d quantized pytorch op to C2

Summary:
Add support for sigmoid in the conversion flow through onnx

Test Plan:
python test/onnx/test_pytorch_onnx_caffe2_quantized.py TestQuantizedOps.test_quantized_sigmoid
python test/onnx/test_pytorch_onnx_caffe2_quantized.py TestQuantizedOps.test_small_model

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D20433680](https://our.internmc.facebook.com/intern/diff/D20433680)

### CODE REVIEW

**Review by z-a-f — APPROVED** [review]
LGTM

File: torch/csrc/jit/passes/onnx/unpack_quantized_weights.cpp [hunk 1]
```diff
@@ -79,6 +79,10 @@ double getScaleFromInput(Node* input_node) {
         "quantized::add expected scale to be 3rd input");
     scale = toIValue(input_node->inputs()[2]);
     return scale.value().toDouble();
+  } else if (input_name == "aten::sigmoid") {
+    // For the _caffe2::Int8Sigmoid op output scale is 1.0/256
+    // And output zero_point is set to 0 (quint8 type).
+    return 1.0 / 256;
```
Comment (houseroad) [inline torch/csrc/jit/passes/onnx/unpack_quantized_weights.cpp]: Nit: use 1.0L to be safe

File: test/onnx/test_pytorch_onnx_caffe2_quantized.py [hunk 2]
```diff
@@ -40,6 +41,30 @@ def forward(self, x):
         x = np.random.random((1, 2)).astype("float32")
         self.generic_test(QModule(op), (x,), input_names=["x"])
 
+    def generic_model_test(self, model, sample_inputs, input_names=None):
```
Comment (houseroad) [inline test/onnx/test_pytorch_onnx_caffe2_quantized.py]: Can we just extend generic_test?

**Review by houseroad — COMMENTED** [review]
Some inline comments

File: test/onnx/test_pytorch_onnx_caffe2_quantized.py [hunk 3]
```diff
@@ -144,7 +157,7 @@ def forward(self, x):
                 return self.dequant(res)
 
         x = np.random.rand(1, 2, 3, 4).astype("float32")
-        self.generic_test(QUpsampleModule(), (x,), input_names=["x"])
+        self.generic_test(QUpsampleModule(), (x,), input_names=["x"], decimal=0)
```
Comment (raghuramank100) [inline test/onnx/test_pytorch_onnx_caffe2_quantized.py]: Suggest using decimal=5 for cases where we expect an exact match.

Comment (supriyar) [reply test/onnx/test_pytorch_onnx_caffe2_quantized.py]: Oops, yeah my bad

**Pull request closed without merging**